App/Services/SettingsManager.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Coder.Desktop.App.Services;
+
+/// <summary>
+/// Settings contract exposing properties for app settings.
+/// </summary>
+public interface ISettingsManager
+{
+    /// <summary>
+    /// Returns the value of the StartOnLogin setting. Returns <c>false</c> if the key is not found.
+    /// </summary>
+    bool StartOnLogin { get; set; }
+
+    /// <summary>
+    /// Returns the value of the ConnectOnLaunch setting. Returns <c>false</c> if the key is not found.
+    /// </summary>
+    bool ConnectOnLaunch { get; set; }
+}
+
+/// <summary>
+/// Implemention of <see cref="ISettingsManager"/> that persists settings to a JSON file
+/// located in the user's local application data folder.
+/// </summary>
+public sealed class SettingsManager : ISettingsManager
+{
+    private readonly string _settingsFilePath;
+    private Settings _settings;
+    private readonly string _fileName = "app-settings.json";
+    private readonly string _appName = "CoderDesktop";
+    private readonly object _lock = new();
+
+    public const string ConnectOnLaunchKey = "ConnectOnLaunch";
+    public const string StartOnLoginKey = "StartOnLogin";
+
+    public bool StartOnLogin
```

Comment:
Since the registry is the source of truth for this, it should probably just be removed from the settings file entirely.